lib/matplotlib/axes/_axes.py

```diff
@@ -3682,6 +3682,11 @@ def apply_mask(arrays, mask):
                     f"'{dep_axis}err' (shape: {np.shape(err)}) must be a "
                     f"scalar or a 1D or (2, n) array-like whose shape matches "
                     f"'{dep_axis}' (shape: {np.shape(dep)})") from None
+            if np.any(err == None):  # noqa: E711
```

Comment:
[speculation, because I’m not near a console, so can’t test] Maybe I‘m overly careful, but I could imagine that comparing a numpy array to a python object is a relatively costly operation.
Since broadcast_to made err a ndarray, we can 
```suggestion
            if err.dtype is object and np.any(err == None):  # noqa: E711
```

Reply:
```python
>>> np.array([2, None, 1])
array([2, None, 1], dtype=object)
>>> a = _
>>> a.dtype
dtype('O')
>>> a.dtype is object
False
>>> np.dtype(object)
dtype('O')
>>> a.dtype is np.dtype(object)
True
>>> a.dtype is np.dtype("O")
True
```

The return value of `err.dtype` is not `object` directly, hence the descrepency, but the idea can be accomplished.

Reply:
Fixed!